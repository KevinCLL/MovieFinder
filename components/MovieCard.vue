<script>
export default {
  props: {
    movie: {
      type: Object,
      default: () => {},
    },
  },
  data() {
    return {
      expandedMovie: null,
      posterNotFound:
        'https://i.pinimg.com/originals/c2/d1/e8/c2d1e8ba4e5b23bbad9bf78b695da39a.jpg',
    }
  },
  async mounted() {
    const expandedMovie = await this.$axios.$get('http://www.omdbapi.com/', {
      params: {
        apikey: '89fef3ea',
        i: this.movie.imdbID,
        plot: 'full',
      },
    })
    this.expandedMovie = expandedMovie
  },
}
</script>

<template>
  <div class="flex flex-col items-center w-full p-3 h-full">
    <div class="border bg-white p-2 h-moviecard relative">
      <div
        v-if="movie.Poster !== 'N/A'"
        class="mb-2 flex items-center h-poster"
      >
        <img class="h-100" :src="movie.Poster" :alt="movie.Title" />
      </div>
      <div
        v-else
        class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-500 to-indigo-700 flex"
      >
        <p class="text-3xl text-white ml-10">Poster not found</p>
      </div>
      <div class="h-5">
        <h2 class="text-2xl">
          <span class="bold">{{ `${movie.Title}` }}</span>
          <span
            v-if="expandedMovie && expandedMovie.Director !== 'N/A'"
            class="italic"
          >
            {{ `by ${expandedMovie.Director}` }}
          </span>
        </h2>
        <h3 class="text-xl">
          {{ movie.Year }}
        </h3>
      </div>
      <img
        class="cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
        src="~/assets/img/magnifying_glass_icon.png"
        alt="magnifying glass"
        @click="$emit('movie-select', expandedMovie)"
      />
    </div>
  </div>
</template>

<style scoped></style>
