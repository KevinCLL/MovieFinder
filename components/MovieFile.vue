<script>
export default {
  name: 'MovieFile',
  props: {
    movie: {
      type: Object,
      default: () => {},
    },
  },
  computed: {
    screenSize() {
      return this.$store.state.windowWidth
    },
    movieInfo() {
      const infoWantedAndSorted = [
        ['Title', this.movie.Title],
        ['Released', this.movie.Released],
        ['Director', this.movie.Director],
        ['Production', this.movie.Production],
        ['Writer', this.movie.Writer],
        ['Actors', this.movie.Actors],
        ['Country', this.movie.Country],
        ['Language', this.movie.Language],
        ['Genre', this.movie.Genre],
        ['Rated', this.movie.Rated],
        ['Awards', this.movie.Awards],
        ['BoxOffice', this.movie.BoxOffice],
        ['Plot', this.movie.Plot],
      ]
      return infoWantedAndSorted.filter((item) => this.itemExists(item[1]))
    },
  },
  methods: {
    itemExists(item) {
      return item && item !== 'N/A'
    },
  },
}
</script>

<template>
  <div
    class="flex w-full p-10 bg-gray-200 relative"
    :class="{ 'flex-col-reverse': screenSize < 763 }"
  >
    <div
      id="movieclose"
      class="cursor-pointer h-icon w-icon absolute right-0 top-0 mt-1 text-xl"
      @click="$emit('close')"
    >
      âœ–
    </div>
    <dl>
      <template v-for="item in movieInfo">
        <dt :key="`${item[0]}-t`">{{ item[0] }}:</dt>
        <dd :key="`${item[0]}-d`">{{ item[1] }}</dd>
      </template>
    </dl>
    <div class="flex flex-col min-w-1/4">
      <div v-if="itemExists(movie.Poster)" class="my-0 mx-auto">
        <img :src="movie.Poster" :alt="movie.Title" />
      </div>
      <div v-if="movie.Ratings && movie.Ratings.length > 0">
        <div v-for="rating in movie.Ratings" :key="rating.Source">
          {{ rating.Source }} : {{ rating.Value }}
        </div>
      </div>
      <div
        v-if="itemExists(movie.imdbID)"
        class="cursor-pointer bg-yellow-400 mb-4 text-center"
      >
        <a :href="`https://www.imdb.com/title/${movie.imdbID}`" target="_blank">
          View IMDB
        </a>
      </div>
    </div>
  </div>
</template>

<style scoped>
dl {
  width: 100%;
  overflow: hidden;
  padding: 0;
  padding-right: 10px;
  margin: 0;
}
dt {
  font-weight: bold;
  float: left;
  text-align: right;
  width: 90px;
  padding-right: 10px;
  margin: 0;
}
dd {
  float: left;
  text-align: left;
  width: 80%;
  padding: 0;
  margin: 0;
}
@media (max-width: 540px) {
  dt {
    text-align: left;
    width: 100%;
  }
  dd {
    width: 100%;
  }
}
</style>
