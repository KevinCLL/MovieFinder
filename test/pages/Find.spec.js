import { mount, createLocalVue } from '@vue/test-utils'
import Find from '@/pages/find/_title.vue'
import MovieCard from '@/components/MovieCard.vue'
import MovieFile from '@/components/MovieFile.vue'
import Vuex from 'vuex'
import VueRouter from 'vue-router'
import VueMeta from 'vue-meta'
import storeConfig from '../store-config'
import detailedMovie from '../data/detailedMovie.js'
import littleDetailsMovie from '../data/littleDetailsMovie.js'
import { good, oneMovie, noMovies } from '../data/movieResults.js'

const localVue = createLocalVue()
localVue.use(VueMeta, { keyName: 'head' })
localVue.use(VueRouter)
localVue.use(Vuex)
localVue.component('movie-card', MovieCard)
localVue.component('movie-file', MovieFile)

describe('Find', () => {
  let wrapper
  let store
  let router

  beforeEach(() => {
    store = new Vuex.Store(storeConfig)
    router = new VueRouter({
      routes: [{ path: '/find/:title', component: Find }],
    })
    router.push('/find/Batman')
    wrapper = mount(Find, {
      localVue,
      mocks: {
        $axios: {
          $get: () => Promise.resolve(detailedMovie),
        },
        $scrollTo: jest.fn(),
      },
      data() {
        return {
          movies: good,
          selectedMovie: null,
          showDetails: false,
        }
      },
      store,
      router,
    })
  })
  afterEach(() => {
    if (wrapper) {
      wrapper.destroy()
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })
  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('has correct <head> content', () => {
    expect(wrapper.vm.$metaInfo.title).toBe('Movie Finder - Batman')
    const descriptionMeta = wrapper.vm.$metaInfo.meta.find(
      (item) => item.hid === 'description'
    )
    expect(descriptionMeta.content).toBe(
      "Not the best movie finder, but maybe it's good enough."
    )
    const noMoviesWrapper = mount(Find, {
      localVue,
      mocks: {
        $axios: {
          $get: () => Promise.resolve(detailedMovie),
        },
      },
      data() {
        return {
          movies: noMovies,
          selectedMovie: null,
          showDetails: false,
        }
      },
      store,
      router,
    })
    expect(noMoviesWrapper.vm.$metaInfo.title).toBe(
      "Movie Finder - Looks like it didn't find it"
    )
    noMoviesWrapper.destroy()
  })

  test('changes selected movie', () => {
    expect(wrapper.vm.$data.selectedMovie).toBe(null)
    wrapper.findAll('.movie-card-glass').at(1).trigger('click')
    expect(wrapper.vm.$data.selectedMovie).toBe(detailedMovie)
  })

  test('closes movie details', async () => {
    wrapper.findAll('.movie-card-glass').at(1).trigger('click')
    expect(wrapper.vm.$data.showDetails).toBe(true)
    await localVue.nextTick()
    wrapper.find('#movieclose').trigger('click')
    expect(wrapper.vm.$data.showDetails).toBe(false)
  })

  test('shows less cards if not enough movies', () => {
    const oneMovieWrapper = mount(Find, {
      localVue,
      mocks: {
        $axios: {
          $get: () => Promise.resolve(littleDetailsMovie),
        },
      },
      data() {
        return {
          movies: oneMovie,
          selectedMovie: null,
          showDetails: false,
        }
      },
      store: new Vuex.Store({
        state: {
          windowWidth: 777,
          screenSize: 'medium',
        },
      }),
      router,
    })
    expect(oneMovieWrapper.vm.cardsPerPage).toBe(1)
    oneMovieWrapper.destroy()
  })

  test("shows poster template if didn't has any poster", () => {
    const oneMovieWrapper = mount(Find, {
      localVue,
      mocks: {
        $axios: {
          $get: () => Promise.resolve(littleDetailsMovie),
        },
      },
      data() {
        return {
          movies: oneMovie,
          selectedMovie: null,
          showDetails: false,
        }
      },
      store,
      router,
    })
    expect(oneMovieWrapper.findAll('.no-poster').at(0).text()).toBe(
      'Poster not found'
    )
    oneMovieWrapper.destroy()
  })

  test("says sorry if didn't find any movie", () => {
    const noMoviesWrapper = mount(Find, {
      localVue,
      mocks: {
        $axios: {
          $get: () => Promise.resolve(detailedMovie),
        },
      },
      data() {
        return {
          movies: noMovies,
          selectedMovie: null,
          showDetails: false,
        }
      },
      store,
      router,
    })
    expect(noMoviesWrapper.find('#sorry').text()).toBe(
      'Sorry, but I have not found any movie with that name'
    )
    noMoviesWrapper.destroy()
  })
})
