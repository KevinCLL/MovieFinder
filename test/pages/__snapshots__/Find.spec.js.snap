// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Find renders correctly 1`] = `
<div
  class="flex flex-col"
>
  <div
    class="VueCarousel"
  >
    <div
      class="VueCarousel-wrapper"
    >
      <div
        class="VueCarousel-inner"
        style="transform: translate(0px, 0); transition: 0.5s ease transform; flex-basis: 0px; visibility: hidden; height: auto;"
      >
        <div
          class="VueCarousel-slide VueCarousel-slide-active"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="Primer"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BNjc3OWVjMTItYjc0Yi00NmFlLTk2YTgtYmU0MzcxMjBkNTYxXkEyXkFqcGdeQXVyMTQxNzMzNDI@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Primer
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        2004
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          class="VueCarousel-slide VueCarousel-slide-active VueCarousel-slide-center"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="Next: A Primer on Urban Painting"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BODk5MDQ1MDI4Ml5BMl5BanBnXkFtZTgwNzI4MDA2MDE@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Next: A Primer on Urban Painting
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        2005
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          class="VueCarousel-slide VueCarousel-slide-active"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="Primer Festival de Mortadelo y Filemón, agencia de información"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BMDk5N2Q0MjAtYjI4ZC00YjdlLWJmYjgtMDNjYzdiZTU1OTA1XkEyXkFqcGdeQXVyODI2MDA4NQ@@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Primer Festival de Mortadelo y Filemón, agencia de información
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        1969
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="Más sabe el diablo: El primer golpe"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BOGZlNzRmM2YtZjAwNy00NDYyLTlkMWQtYmY4ZTMyZWM2MDlmXkEyXkFqcGdeQXVyMjQ2MDQwNDU@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Más sabe el diablo: El primer golpe
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        2010
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-700 to-blue-900 flex"
              >
                <p
                  class="no-poster text-3xl text-white my-0 mx-auto"
                >
                  
        Poster not found
      
                </p>
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Primer Feliks Langus ali Kako ujeti svobodo
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        1991
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-700 to-blue-900 flex"
              >
                <p
                  class="no-poster text-3xl text-white my-0 mx-auto"
                >
                  
        Poster not found
      
                </p>
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Primer amor... tres años después
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        2001
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="El primer cuartel"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BZDVhNWM5ZDItMzkyYi00ZGQ5LWEyNGMtYTg2ZmVlOTYyNjg2L2ltYWdlXkEyXkFqcGdeQXVyNTUyMTMyMDg@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    El primer cuartel
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        1966
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-700 to-blue-900 flex"
              >
                <p
                  class="no-poster text-3xl text-white my-0 mx-auto"
                >
                  
        Poster not found
      
                </p>
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    Primer plano
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        2001–
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex h-poster"
              >
                <img
                  alt="The Pimp Primer"
                  class="h-100 my-0 mx-auto"
                  src="https://m.media-amazon.com/images/M/MV5BNTRmOTJkOTItNTI1ZC00ZTk0LTg1MmQtZGYzZTM3NDE4NGY1XkEyXkFqcGdeQXVyNjg5Nzc3NTU@._V1_SX300.jpg"
                />
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    The Pimp Primer
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        1970
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
        <div
          aria-hidden="true"
          class="VueCarousel-slide"
          role="tabpanel"
          tabindex="-1"
        >
          <div
            class="flex flex-col items-center w-full p-3 h-full"
          >
            <div
              class="border bg-white p-2 h-moviecard w-moviecard relative"
            >
              <div
                class="mb-2 flex items-center h-poster bg-gradient-to-tr from-blue-700 to-blue-900 flex"
              >
                <p
                  class="no-poster text-3xl text-white my-0 mx-auto"
                >
                  
        Poster not found
      
                </p>
              </div>
               
              <div
                class="h-5 text-center"
              >
                <h2>
                  <span
                    class="text-2xl"
                  >
                    A Communications Primer
                  </span>
                   
                  <span
                    class="italic text-xl"
                  >
                    
          by Shane Carruth
        
                  </span>
                </h2>
                 
                <h3
                  class="text-xl"
                >
                  
        1953
      
                </h3>
              </div>
               
              <img
                alt="magnifying glass"
                class="movie-card-glass cursor-pointer h-icon w-icon absolute right-0 bottom-0 m-3"
                src="~/assets/img/magnifying_glass_icon.png"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <!---->
     
    <div
      class="VueCarousel-pagination"
      data-v-438fd353=""
      style=""
    >
      <div
        class="VueCarousel-dot-container"
        data-v-438fd353=""
        role="tablist"
        style="margin-top: 20px;"
      >
        <button
          aria-hidden="false"
          aria-label="Item 0"
          aria-selected="true"
          class="VueCarousel-dot VueCarousel-dot--active"
          data-v-438fd353=""
          role="tab"
          style="margin-top: 20px; padding: 10px; width: 10px; height: 10px; background-color: rgb(0, 0, 0);"
          title="Item 0"
          value="Item 0"
        />
        <button
          aria-hidden="false"
          aria-label="Item 1"
          aria-selected="false"
          class="VueCarousel-dot"
          data-v-438fd353=""
          role="tab"
          style="margin-top: 20px; padding: 10px; width: 10px; height: 10px; background-color: rgb(239, 239, 239);"
          title="Item 1"
          value="Item 1"
        />
        <button
          aria-hidden="false"
          aria-label="Item 2"
          aria-selected="false"
          class="VueCarousel-dot"
          data-v-438fd353=""
          role="tab"
          style="margin-top: 20px; padding: 10px; width: 10px; height: 10px; background-color: rgb(239, 239, 239);"
          title="Item 2"
          value="Item 2"
        />
        <button
          aria-hidden="false"
          aria-label="Item 3"
          aria-selected="false"
          class="VueCarousel-dot"
          data-v-438fd353=""
          role="tab"
          style="margin-top: 20px; padding: 10px; width: 10px; height: 10px; background-color: rgb(239, 239, 239);"
          title="Item 3"
          value="Item 3"
        />
      </div>
    </div>
  </div>
   
  <div
    class="my-10"
    id="expanded"
  >
    <!---->
  </div>
   
  <div
    class="text-center w-full"
  >
    <div
      class="inline cursor-pointer h-icon w-icon text-3xl text-white"
    >
      
      ↑
    
    </div>
  </div>
</div>
`;
